From d05621d33f7366f7960458ec67e6163126e57b50 Mon Sep 17 00:00:00 2001
From: Cleiton Bueno <cleitonrbueno@gmail.com>
Date: Fri, 19 Feb 2021 12:30:56 -0300
Subject: [PATCH] re-patch [resolv.conf.auto]: Use /run instead of /tmp

Using netifd as network manager along with systemd provokes the bad usage
of path, in this case /tmp/. All the runtime configuration needs to be
sotored as part of /run

Problem:
========
assuming: ln -fs /tmp/resolv.conf.auto /etc/resolv.conf

If any application tend to use Filesystem namespace to have separate /tmp and
try to use DNS resolution from netifd, it will not be able to resolve the path/symlink.

Usage of PrivateTmp=yes:
------------------------
For example, systemd-timesyncd.service in systemd uses the option "PrivateTmp=yes"
in the service file to have separate /tmp filesystem namespace and uses glibc
getaddrinfo for name resolution, which in turm follows the symlink to /tmp which
doesn't exit in new namespace

So moving the runtime configuration to /run here

Signed-off-by: Parthiban Nallathambi <pn@denx.de>
Signed-off-by: Alejandro Hernandez <aehs29@gmail.com>

---
 netifd.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/netifd.h b/netifd.h
index a4a146c..94a57b5 100644
--- a/netifd.h
+++ b/netifd.h
@@ -32,12 +32,12 @@
 #define DEFAULT_MAIN_PATH	"./examples"
 #define DEFAULT_CONFIG_PATH	"./config"
 #define DEFAULT_HOTPLUG_PATH	"./examples/hotplug-cmd"
-#define DEFAULT_RESOLV_CONF	"./tmp/resolv.conf"
+#define DEFAULT_RESOLV_CONF	"./run/resolv.conf"
 #else
 #define DEFAULT_MAIN_PATH	"/lib/netifd"
 #define DEFAULT_CONFIG_PATH	NULL /* use the default set in libuci */
 #define DEFAULT_HOTPLUG_PATH	"/sbin/hotplug-call"
-#define DEFAULT_RESOLV_CONF	"/tmp/resolv.conf.d/resolv.conf.auto"
+#define DEFAULT_RESOLV_CONF	"/run/resolv.conf.d/resolv.conf.auto"
 #endif
 
 extern const char *resolv_conf;
-- 
2.7.4

